src/lib/datepicker/datepicker.spec.ts

```diff
@@ -95,14 +127,110 @@ describe('MdDatepicker', () => {
       expect(() => testComponent.datepicker.openStandardUi()).toThrow();
     });
   });
+
+  describe('datepicker with startAt', () => {
+    let fixture: ComponentFixture<DatepickerWithStartAt>;
+    let testComponent: DatepickerWithStartAt;
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(DatepickerWithStartAt);
+      fixture.detectChanges();
+
+      testComponent = fixture.componentInstance;
+    });
+
+    it('explicit startAt should override input value', () => {
+      expect(testComponent.datepicker.startAt).toEqual(new SimpleDate(2010, 0, 1));
+    });
+  });
+
+  describe('datepicker with ngModel', () => {
+    let fixture: ComponentFixture<DatepickerWithNgModel>;
+    let testComponent: DatepickerWithNgModel;
+
+    beforeEach(fakeAsync(() => {
+      fixture = TestBed.createComponent(DatepickerWithNgModel);
+      detectModelChanges(fixture);
+
+      testComponent = fixture.componentInstance;
+    }));
+
+    it('should update datepicker when model changes', fakeAsync(() => {
+      expect(testComponent.datepickerInput.value).toBeNull();
+      expect(testComponent.datepicker.selected).toBeNull();
+
+      let selected = new SimpleDate(2017, 0, 1);
+      testComponent.selected = selected;
+      detectModelChanges(fixture);
+
+      expect(testComponent.datepickerInput.value).toEqual(selected);
+      expect(testComponent.datepicker.selected).toEqual(selected);
+    }));
+
+    it('should update model when date is selected', fakeAsync(() => {
+      expect(testComponent.selected).toBeNull();
+      expect(testComponent.datepickerInput.value).toBeNull();
+
+      let selected = new SimpleDate(2017, 0, 1);
+      testComponent.datepicker.selected = selected;
+      detectModelChanges(fixture);
+
+      expect(testComponent.selected).toEqual(selected);
+      expect(testComponent.datepickerInput.value).toEqual(selected);
+    }));
```

Comment:
We should add tests that ensure that touched, disabled, and dirty still work, given that we're re-implementing them in the value accessor